File: productos-frontend/src/app/models/cliente-fidelidad.dto.ts
import { CuponDTO } from './cupon.model'; 

export interface ClienteFidelidadDTO {
    clienteId: number;
    nombreCompleto: string;
    puntosAcumulados: number;
    equivalenciaMonetaria: string;
    cuponesDisponibles: CuponDTO[];
}

File: productos/src/main/java/com/masterserv/productos/dto/ClienteFidelidadDTO.java
package com.masterserv.productos.dto;

import lombok.Data;
import java.math.BigDecimal;
import java.util.List;

@Data
public class ClienteFidelidadDTO {
    private Long clienteId;
    private String nombreCompleto;
    
    // Puntos
    private Integer puntosAcumulados;
    private String equivalenciaMonetaria; // Ej: "equivale a $5000"
    
    // Cupones listos para usar
    private List<CuponDTO> cuponesDisponibles; 
}